addons/mail/controllers/discuss/public_page.py

```diff
@@ -52,10 +52,19 @@ def discuss_channel_invitation(self, channel_id, invitation_token):
     @http.route("/discuss/channel/<int:channel_id>", methods=["GET"], type="http", auth="public")
     @add_guest_to_context
     def discuss_channel(self, channel_id):
-        channel = request.env["discuss.channel"].search([("id", "=", channel_id)])
+        # sudo(): discuss.channel - checking if the channel exists
+        channel = request.env["discuss.channel"].sudo().search([("id", "=", channel_id)])
```

Comment:
I'd say the same here, I don't think this `sudo()` call is legit: why would we find a channel the user doesn't have access to? And the check below looks very fragile: we are either logged in, or we have a guest. But this does not take into account guest channels nor user groups.